src/hotspot/cpu/riscv/assembler_riscv.cpp

```diff
@@ -116,6 +116,31 @@ void Assembler::_li(Register Rd, int64_t imm) {
   }
 }
 
+void Assembler::li(Register Rd, int64_t imm, int32_t &offset) {
```

Comment:
It looks to me that we don't need to introduce another 'li' function here. I think you can mask out the lower 12bits of the address and move the result to register with existing 'mv' function. And keep the lower 12bits in offset at the same time.

Reply:
Thanks for the advice, very reasonable. I have changed that (need further testing). Others all done.